pfcpiface/main.go

```diff
@@ -25,13 +34,97 @@ type pdr struct {
 	proto    uint8
 }
 
-func addPDR(ctx context.Context, c pb.BESSControlClient, pdr pdr) {
+type Conf struct {
+	Mode             string      `json:"mode"`
+	UECidr           string      `json:"ue_cidr"`
+	EnbCidr          string      `json:"enb_cidr"`
+	MaxSessions      uint32      `json:"max_sessions"`
+	MaxIPDefragFlows uint32      `json:"max_ip_defrag_flows"`
+	IPFragWithEthMtu uint32      `json:"ip_frag_with_eth_mtu"`
+	Measure          bool        `json:"measure"`
+	S1UIntf          Interface   `json:"s1u"`
+	SGIIntf          Interface   `json:"sgi"`
+	Workers          uint32      `json:"workers"`
+	Cpiface          CpifaceType `json:"cpiface"`
+}
+
+type Interface struct {
+	Name string `json:"ifname"`
+	Nat  string `json:"ip_masquerade"`
+}
+
+type CpifaceType struct {
+	N3_IP         string `json:"s1u_sgw_ip"`
+	Northbound_IP string `json:"zmqd_nb_ip"`
+	Local_IP      string `json:"zmqd_ip"`
+	Hostname      string `json:"hostname"`
+}
```

Comment:
you can limit it to the things you care about for this program
```
type Conf struct {
        MaxSessions      uint32      `json:"max_sessions"`
	Cpiface          CpifaceType `json:"cpiface"`
}

type CpifaceType struct {
	N3_IP         string `json:"s1u_sgw_ip"`
}
```

Reply:
you can even create an new block json for pfcp instead of reusing cpiface. minimal confusion (zmq n all)